src/Analyzers/CSharp/Analyzers/UsePatternMatching/UsePatternMatchingHelpers.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternMatching
+{
+    internal static class UsePatternMatchingHelpers
+    {
+        public static bool TryGetPartsOfAsAndMemberAccessCheck(
+            BinaryExpressionSyntax asExpression,
+            [NotNullWhen(true)] out ConditionalAccessExpressionSyntax? conditionalAccessExpression,
+            out BinaryExpressionSyntax? binaryExpression,
+            out IsPatternExpressionSyntax? isPatternExpression,
+            out LanguageVersion requiredLanguageVersion)
+        {
+            conditionalAccessExpression = null;
+            binaryExpression = null;
+            isPatternExpression = null;
+            requiredLanguageVersion = LanguageVersion.CSharp8;
+
+            if (asExpression.Kind() == SyntaxKind.AsExpression)
```

Comment:
nit - maybe
```
if (asExpression.Kind() != SyntaxKind.AsExpression)
{
   return false;
}
```

Reply:
yeah, that's better.  will do in followup :)